### PR TITLE
Minor fixes to make Jenkins happy again.

### PR DESCRIPTION
**Pull request by purbon**
Minor fixes after logstash-core gem release.
 * mocking out the patterns directory.
 * devutils dependency deduplication.

### CODE REVIEW

File: spec/filters/grok_spec.rb [hunk 1]
```diff
@@ -665,6 +666,7 @@
 
     after do
       @file.unlink
+      FileUtils.rm_rf(pattern_dir)
```
Comment (jsvd) [inline spec/filters/grok_spec.rb]: I guess we should only remove it if we know that it was created during the test?

File: spec/filters/grok_spec.rb [hunk 2]
```diff
@@ -648,8 +647,10 @@
     require 'tmpdir'
     require 'tempfile'
 
+    let(:pattern_dir) { File.join(LogStash::Environment::LOGSTASH_HOME, "patterns") }
+
     before do
-      pattern_dir = File.join(LogStash::Environment::LOGSTASH_HOME, "patterns")
+      FileUtils.mkdir(pattern_dir) unless File.exist?(pattern_dir)
```
Comment (jsvd) [inline spec/filters/grok_spec.rb]: Dir.exist?

File: spec/filters/grok_spec.rb (hunk 1 above)
Comment (purbon) [inline spec/filters/grok_spec.rb]: makes sense

File: spec/filters/grok_spec.rb (hunk 2 above)
Comment (purbon) [inline spec/filters/grok_spec.rb]: I think is basically doing the same, but will change ... semantically speaking makes sense.

**Comment by electrical** [comment]
LGTM

**Pull request closed without merging**